packages/ui/src/primitives/tailwind/Select/index.tsx

```diff
@@ -23,183 +23,281 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
 Infinite Reality Engine. All Rights Reserved.
 */
 
-import React, { useEffect, useRef } from 'react'
-import { useTranslation } from 'react-i18next'
-import { MdOutlineKeyboardArrowDown } from 'react-icons/md'
-import { twMerge } from 'tailwind-merge'
-
-import { calculateAndApplyYOffset } from '@ir-engine/common/src/utils/offsets'
 import { useClickOutside } from '@ir-engine/common/src/utils/useClickOutside'
-import { useHookstate } from '@ir-engine/hyperflux'
-
-import Input, { InputProps } from '../Input'
-
-export type OptionValueType = string | number
-
-export type SelectOptionsType = { label: string; value: any; disabled?: boolean }
+import { ChevronDownSm, HelpIconSm } from '@ir-engine/ui/src/icons'
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { DropdownItem } from '../Dropdown'
+import { InputProps, variantSizes } from '../Input'
+import Tooltip from '../Tooltip'
 
-export interface SelectProps<T extends OptionValueType> {
-  label?: string
+export interface OptionType {
+  value: string | number
+  label: string
+  Icon?: ({ className }: { className?: string }) => JSX.Element
+  /**text shown on the right end */
+  secondaryText?: string
+  disabled?: boolean
+  selected?: boolean
   className?: string
-  error?: string
-  description?: string
-  options: SelectOptionsType[]
-  currentValue: T
+}
+
+export interface SelectProps<T = string | number> {
+  options: OptionType[]
+  width?: 'sm' | 'md' | 'lg' | 'full'
+  inputSizeVariant?: InputProps['variantSize']
   onChange: (value: T) => void
-  placeholder?: string
+  value: T
+  renderValue?: (value: T) => string
```

Comment:
lets add some docstring comments for `renderValue` which will help other to understand its use

was thinking about its use case - currently we have it not utilized it anywhere. in most (if not all) cases, the person using `select` will use `.map` to fill the option's label

Reply:
removed it